process/sync/shardblock.go

```diff
@@ -132,7 +134,16 @@ func (boot *ShardBootstrap) StartSyncingBlocks() {
 // These methods will execute the block and its transactions. Finally if everything works, the block will be committed
 // in the blockchain, and all this mechanism will be reiterated for the next block.
 func (boot *ShardBootstrap) SyncBlock() error {
-	return boot.syncBlock()
+	err := boot.syncBlock()
+	isErrGetNodeFromDB := err != nil && strings.Contains(err.Error(), core.GetNodeFromDBErrorString)
```

Comment:
extract L138-L144 to a function `syncAccountsIfNeeded(err error)`

Reply:
extraction does not work here. it complicates things. needs internal interface and all.